src/hmem_rocr.c

```diff
@@ -856,6 +889,13 @@ int rocr_hmem_init(void)
 	if (ret)
 		goto fail;
 
+	memset(&rocr_agents, 0, sizeof(rocr_agents));
+
+	hsa_ret = ofi_hsa_iterate_agents(rocr_hsa_agent_callback, NULL);
+	if (hsa_ret != HSA_STATUS_SUCCESS &&
+	    hsa_ret != HSA_STATUS_INFO_BREAK)
+	    goto fail;
+
```

Comment:
I think the patch is missing a piece of code in rocr_hmem_dl_init(), something like

   hsa_ops.hsa_iterate_agents = dlsym(hsa_handle, "hsa_iterate_agents");
        if (!hsa_ops.hsa_iterate_agents) {
                FI_WARN(&core_prov, FI_LOG_CORE,
                        "Failed to find hsa_iterate_agents\n");
                goto err;
        }